spec/connection_spec.rb

```diff
@@ -149,8 +149,9 @@
       ]
       headers = []
       @conn.on(:frame) do |bytes|
-        # bytes[3]: frame's type field
-        headers << f.parse(bytes) if [1, 5, 9].include?(bytes[3].ord)
+        # buf[3]: frame's type field
+        buf = Buffer.new(bytes)
```

Comment:
Perhaps we should pass in a buffer object in the callback? Manually converting to a buffer each time seems like a gotcha / unnecessary complication.